tests/test_resource_tracker.py

```diff
@@ -38,45 +44,54 @@ def test_child_retrieves_resource_tracker(self):
         # child process. If the two processes do not share the same
         # resource_tracker, a cache KeyError should be printed in stderr.
         import subprocess
-        folder_name = 'loky_tempfolder'
         cmd = '''if 1:
         import os, sys
 
         from loky import ProcessPoolExecutor
         from loky.backend import resource_tracker
         from loky.backend.semlock import SemLock
+        from tempfile import NamedTemporaryFile
+        import sys
 
+
+        tmpfile = NamedTemporaryFile(delete=False)
+        tmpfile.close()
+        filename = tmpfile.name
         resource_tracker.VERBOSE=True
-        folder_name = "{}"
 
-        # We don't need to create the semaphore as registering / unregistering
-        # operations simply add / remove entries from a cache, but do not
-        # manipulate the actual semaphores.
-        resource_tracker.register(folder_name, "folder")
+        resource_tracker.register(filename, "file")
 
-        def unregister(name, rtype):
-            # resource_tracker.unregister is actually a bound method of the
+        def maybe_unlink(name, rtype):
+            # resource_tracker.maybe_unlink is actually a bound method of the
             # ResourceTracker. We need a custom wrapper to avoid object
             # serialization.
             from loky.backend import resource_tracker
-            resource_tracker.unregister(folder_name, rtype)
+            resource_tracker.maybe_unlink(name, rtype)
 
+        sys.stdout.write(filename + "\\n")
+        sys.stdout.flush()
         e = ProcessPoolExecutor(1)
-        e.submit(unregister, folder_name, "folder").result()
+        e.submit(maybe_unlink, filename, "file").result()
         e.shutdown()
         '''
         try:
             p = subprocess.Popen(
-                [sys.executable, '-E', '-c', cmd.format(folder_name)],
-                stderr=subprocess.PIPE)
+                [sys.executable, '-E', '-c', cmd],
+                stderr=subprocess.PIPE,
+                stdout=subprocess.PIPE)
             p.wait()
 
+            filename = p.stdout.readline().decode('utf-8').strip()
             err = p.stderr.read().decode('utf-8')
             p.stderr.close()
+            p.stdout.close()
+
 
-            assert re.search("unregister %s" % folder_name, err) is not None
+            pattern = r"decremented refcount of file %s" % filename
+            assert re.search(_escape(pattern), err) is not None
             assert re.search("leaked", err) is None
-            assert re.search("KeyError: '%s'" % folder_name, err) is None
+            assert re.search(
+                    r"KeyError: '%s'" % _escape(filename), err) is None
```

Comment:
To avoid having to escape backslashed in filenames you could just do:


```python
pattern = "decremented refcount of file %s" % filename
assert pattern in err
```

The same remark applies for the other lines of this block which could be simplified similarly. The pytest failure message would also become more informative.